hw/xbox/xid-sdl.c

```diff
@@ -512,14 +512,16 @@ static void usb_xbox_gamepad_realize(USBDevice *dev, Error **errp)
 #endif
 
 #ifdef FORCE_FEEDBACK
-    s->sdl_haptic = SDL_HapticOpenFromJoystick(s->sdl_gamepad);
+    s->sdl_haptic = SDL_HapticOpen(s->device_index);
     if (s->sdl_haptic == NULL) {
         fprintf(stderr, "Joystick doesn't support haptic\n");
     } else {
         if ((SDL_HapticQuery(s->sdl_haptic) & SDL_HAPTIC_LEFTRIGHT) == 0) {
             fprintf(stderr, "Joystick doesn't support necessary haptic effect\n");
```

Comment:
Error message should be more specific, so users can debug this issue + so we don't have to rename it when adding additional checks `[...] effect (SDL_HAPTIC_LEFTRIGHT)` or something